scipy/stats/stats.py

```diff
@@ -223,16 +224,20 @@
            'somersd']
 
 
-def _contains_nan(a, nan_policy='propagate'):
+def _contains_nan(a, nan_policy='propagate', summary_check=True):
     policies = ['propagate', 'raise', 'omit']
     if nan_policy not in policies:
         raise ValueError("nan_policy must be one of {%s}" %
                          ', '.join("'%s'" % s for s in policies))
     try:
-        # Calling np.sum to avoid creating a huge array into memory
-        # e.g. np.isnan(a).any()
-        with np.errstate(invalid='ignore'):
-            contains_nan = np.isnan(np.sum(a))
+        # The summary check avoids creating a (potentially huge) array.
+        # However, it will set contains_nan to True for (e.g.) [-inf, 0, +inf].
```

Comment:
I do not observe this behavior:
```python3
a = np.random.rand(10)
a[0] = np.nan
print(np.isnan(np.sum(a)))  # True
a[0] = np.inf
print(np.isnan(np.sum(a)))  # False
a[0] = -np.inf
print(np.isnan(np.sum(a)))  # False
a[0] = 0
print(np.isnan(np.sum(a)))  # False
```
If I am missing the point, please clarify.
Also, it seems to me that whether `summary_check` is `True` or `False`, some sort of "summary check" for NaNs is performed. Is there a more descriptive name for this variable?

Reply:
The test cases you run are not detecting the relevant scenario: when the array contains *both* `+inf` and `-inf`, in which case the two methods differ.

I have clarified the comments; in particular reworded "summary" to "summation"

Reply:
I see.